cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/StructuredRecordWritable.java

```diff
@@ -69,7 +73,13 @@ public void readFields(DataInput in) throws IOException {
     byte[] schemaBytes = new byte[schemaLen];
```

Comment:
Shouldn't you cache based on the `schemaBytes` to save the decoding bytes into string?

Reply:
Fixed. Since there will be limited number of schema, I have used `TreeMap` with `ByteComparator`. Another approach would be to wrap `byte[]` in another class which overrides `equals` and `hashcode` methods